addons/bus/static/src/js/crosstab_bus.js

```diff
@@ -0,0 +1,355 @@
+        var heartbeatKillOld = new Date().getTime() - this.HEARTBEAT_KILL_OLD_PERIOD;
+        var newMaster;
+        for (var peerName in peers) {
+            //check for dead peers
+            if (peers[peerName] < heartbeatKillOld) {
+                continue;
+            }
+            newMaster = peerName;
+            break;
+        }
+
+        if (newMaster === this._id) {
+            //we're next in queue. Electing as master
+            this.lastHeartbeat = new Date().getTime();
+            this._callLocalStorage('setItem', 'heartbeat', this.lastHeartbeat);
+            this._isMasterTab = true;
+            this.startPolling();
+            this.trigger('become_master');
+
+            //removing master peer from queue
+            delete peers[newMaster];
+            this._callLocalStorage('setItem', 'peers', peers);
+        }
+    },
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+    /**
+     * @override
+     */
+    _onFocusChange: function (params) {
+        this._super.apply(this, arguments);
+        this._callLocalStorage('setItem', 'focus', params.focus);
+    },
+    /**
+     * If it's the master tab, the notifications ares broadcasted to other tabs by the
+     * local storage.
+     *
+     * @override
+     */
+    _onPoll: function (notifications) {
+        var notifs = this._super(notifications);
+        if (this._isMasterTab && notifs.length) {
+            this._callLocalStorage('setItem', 'last', this._lastNotificationID);
+            this._callLocalStorage('setItem', 'last_ts', new Date().getTime());
+            this._callLocalStorage('setItem', 'notification', notifs);
+        }
+    },
+    /**
+     * Handler when the local storage is updated
+     *
+     * @private
+     * @param  {jQueryEvent} event
+     * @param  {StorageEvent} event.originalEvent
+     */
+    _onStorage: function (e) {
+        var self = this;
+        var value = JSON.parse(e.newValue);
+        var key = e.key;
+
+        if (this._isRegistered && key === this._generateKey('peers') && !value) {
+            //master was unloaded
+            this._startElection();
+        }
+
+        // last notification id changed
+        if (key === this._generateKey('last')) {
+            this._lastNotificationID = value;
+        }
+        // notifications changed
+
+        if (key === this._generateKey('notification')) {
+            if (!this._isMasterTab) {
+                this.trigger("notification", value);
+            }
+            return;
+        }
+        // update channels
+        if (key === this._generateKey('channels')) {
+            var channels = value;
+            _.each(_.difference(this._channels, channels), function (channel) {
+                self.deleteChannel(channel);
+            });
+            _.each(_.difference(channels, this._channels), function (channel) {
```

Comment:
`_.each(_.difference(channels, this._channels), this.addChannel.bind(this));`

Also you can remove `var self` at the top of the method. :-)